core/src/main/scala/caliban/validation/Validator.scala

```diff
@@ -561,6 +561,7 @@ object Validator {
             types.filterNot(isObject).map(_.name.getOrElse("")).filterNot(_.isEmpty).mkString("", ", ", "."),
           s"The member types of a Union type must all be Object base types."
         )
+      // shouldn't we be validating each object here?
       case _ => IO.unit
```

Comment:
@ghostdogpr just noticed `validateUnion` doesn't check if each object member of the union is valid. We should run `validateObject` on the object members, right?

Reply:
I think they are already validated, because we run the validation on the list of all types that we build recursively.